pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -199,32 +223,238 @@ pub mod pallet {
-			todo!()
+		pub fn force_start_refund(origin: OriginFor<T>, pool_id: T::PoolId) -> DispatchResult {
+			ensure_root(origin)?;
+
+			Self::do_start_refund(pool_id, None)
 		}
 
-		// todo: check if we really need that tx.
 		#[pallet::call_index(8)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn destroy_accounts(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn refund_account(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			account: AccountIdLookupOf<T>,
+			asset_idx: u32,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			let who = T::Lookup::lookup(account)?;
+
+			let pool_details = Pools::<T>::get(&pool_id).ok_or(Error::<T>::PoolUnknown)?;
+
+			ensure!(pool_details.state.is_refunding(), Error::<T>::NotRefunding);
+
+			// get asset id from linked assets vector
+			let asset_id: &FungiblesAssetIdOf<T> = pool_details
+				.bonded_currencies
+				.get(asset_idx.saturated_into::<usize>())
+				.ok_or(Error::<T>::IndexOutOfBounds)?;
+
+			let pool_account = pool_id.clone().into();
+
+			let total_collateral_issuance = Self::get_pool_collateral(&pool_account);
+
+			// nothing to distribute; refunding is complete, user should call start_destroy
+			ensure!(!total_collateral_issuance.is_zero(), Error::<T>::ZeroAmount);
+
+			// TODO: remove any existing locks on the account prior to burning
+
+			// With amount = max_value(), this trait implementation burns the reducible balance on the account and returns the actual amount burnt
+			let burnt = T::Fungibles::burn_from(
+				asset_id.clone(),
+				&who,
+				Bounded::max_value(),
+				Precision::BestEffort,
+				Fortitude::Force,
+			)?;
+
+			// no funds available to be burnt on account; nothing to do here
+			ensure!(!burnt.is_zero(), Error::<T>::ZeroAmount); // TODO: fail or return?
+
+			let sum_of_issuances = pool_details
+				.bonded_currencies
+				.into_iter()
+				.fold(FungiblesBalanceOf::<T>::zero(), |sum, id| {
+					sum.saturating_add(T::Fungibles::total_issuance(id))
+				});
+
+			let amount = burnt
+				.checked_mul(&total_collateral_issuance)
+				.ok_or(ArithmeticError::Overflow)? // TODO: do we need a fallback if this fails?
+				.checked_div(&sum_of_issuances)
+				.ok_or(Error::<T>::ZeroAmount)?; // should be impossible - how would we be able to burn funds if the sum of total supplies is 0?
+
+			if amount.is_zero()
+				|| T::CollateralCurrency::can_deposit(T::CollateralAssetId::get(), &who, amount, Provenance::Extant)
+					.into_result()
+					.is_err()
+			{
+				// funds are burnt but the collateral received is not sufficient to be deposited to the account
+				// this is tolerated as otherwise we could have edge cases where it's impossible to refund at least some accounts
+				return Ok(());
+			}
+
+			let transferred = T::CollateralCurrency::transfer(
+				T::CollateralAssetId::get(),
+				&pool_account,
+				&who,
+				amount,
+				Preservation::Expendable,
```

Comment:
I don't think we're planning on putting any creator deposit into the pool account, so this is expendable. It's fine if this account dies before the pool does IMO

Reply:
I am referring to this: https://github.com/KILTprotocol/kilt-node/blob/341ddf8a33ef12d53dede9f9adca2821ac4737a3/pallets/pallet-bonded-coins/src/lib.rs#L230.

Reply:
And this is a necessary step to bring the balance of the pool account to ED, after which other stuff can be transferred to it.

Reply:
Hence this should only ever be returned when the pool is destroyed, otherwise if the pool account goes below ED, all of the other currencies it holds are cleaned up as well.

Reply:
If I understand touch right it's a no-op for native currencies and for the non-native ones which we primarily target (say, USDT) this would reserve a deposit in native currency (KILT) and thus would not be affected by a transfer of USDT. It's also the touch why I think we don't need to worry about EDs at all when it comes to the collateral. I didn't figure out how to un-touch though via the fungibles trait(s), but worst case the pool creator just has to un-reserve/un-touch manually.

Reply:
Ok yeah I just looked into how the trait is implemented by the fungibles pallet. We take the deposit elsewhere in native currency (the pool manager) so that we can still have an entry for the subject (the pool account) in the collateral currency. That means that as a starting point, the pool account will have `0` balance, and if someone bonds `2`, it will have a balance of `2`. Now, the balance cannot go below `1` as the minimum balance, so having expendable is actually the right choice. I am also not sure how the native currency that is reserved for the touch can be unreserved, but how can the creator do that manually?

Reply:
> how can the creator do that manually?

That would be the refund extrinsic called on the fungibles pallet, https://docs.rs/pallet-assets/latest/src/pallet_assets/lib.rs.html#1621